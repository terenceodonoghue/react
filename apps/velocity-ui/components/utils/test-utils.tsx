/* eslint-disable import/no-extraneous-dependencies */
import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider, {
  ThemeProviderProps,
} from '~/components/core/ThemeProvider';

const Provider: FunctionComponent<ThemeProviderProps> = ({ children }) => (
  <ThemeProvider>{children}</ThemeProvider>
);

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Provider, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
