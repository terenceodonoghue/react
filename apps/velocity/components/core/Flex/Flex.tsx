import React, { FunctionComponent, HTMLAttributes } from 'react';

const Flex: FunctionComponent<HTMLAttributes<HTMLDivElement>> = (props) => (
  <div css={{ display: 'flex' }} data-testid="flex" {...props} />
);

export default Flex;
