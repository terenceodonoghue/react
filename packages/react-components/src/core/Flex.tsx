/** @jsxImportSource @emotion/react */
import { Property } from 'csstype';
import { FunctionComponent, HTMLAttributes } from 'react';

export interface FlexProps extends HTMLAttributes<HTMLDivElement> {
  direction?: Property.FlexDirection;
  wrap?: Property.FlexWrap;
  gap?: Property.Gap;
  columnGap?: Property.ColumnGap;
  rowGap?: Property.RowGap;
  alignItems?: Property.AlignItems;
  justifyContent?: Property.JustifyContent;
  justifyItems?: Property.JustifyItems;
}

const Flex: FunctionComponent<FlexProps> = ({
  alignItems,
  children,
  columnGap,
  direction,
  gap,
  justifyContent,
  rowGap,
  wrap,
  ...props
}) => {
  return (
    <div
      css={{
        display: 'flex',
        flexDirection: direction,
        flexWrap: wrap,
        alignItems,
        justifyContent,
        columnGap,
        rowGap,
        gap,
      }}
      {...props}
    >
      {children}
    </div>
  );
};

export default Flex;
