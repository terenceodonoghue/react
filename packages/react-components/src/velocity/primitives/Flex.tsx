/** @jsxImportSource @emotion/react */
import { CSSProperties, ComponentPropsWithoutRef } from 'react';

import mq, { MediaQuery } from '../utils/mq.js';

export interface FlexProps extends ComponentPropsWithoutRef<'div'> {
  align?: MediaQuery<CSSProperties['alignItems']>;
  direction?: MediaQuery<CSSProperties['flexDirection']>;
  justify?: MediaQuery<CSSProperties['justifyContent']>;
  marginX?: MediaQuery<number | CSSProperties['marginInline']>;
  marginY?: MediaQuery<number | CSSProperties['marginBlock']>;
  paddingX?: MediaQuery<number | CSSProperties['paddingInline']>;
  paddingY?: MediaQuery<number | CSSProperties['paddingBlock']>;
  spacing?: MediaQuery<number | CSSProperties['gap']>;
}

const Flex = ({
  align,
  direction,
  justify,
  marginX,
  marginY,
  paddingX,
  paddingY,
  spacing,
  ...props
}: FlexProps) => (
  <div
    css={mq({
      display: 'flex',
      flexDirection: direction,
      alignItems: align,
      justifyContent: justify,
      gap: spacing,
      marginBlock: marginY,
      marginInline: marginX,
      paddingBlock: paddingY,
      paddingInline: paddingX,
    })}
    {...props}
  />
);

export default Flex;
