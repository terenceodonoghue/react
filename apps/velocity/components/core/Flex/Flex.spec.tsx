import React from 'react';
import { faker } from '@faker-js/faker';
import { render } from 'test-utils';
import Flex from './Flex';

describe('Flex', () => {
  it('renders content', () => {
    const content = faker.hacker.phrase();
    const { container, getByTestId } = render(<Flex>{content}</Flex>);

    expect(getByTestId('flex')).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
