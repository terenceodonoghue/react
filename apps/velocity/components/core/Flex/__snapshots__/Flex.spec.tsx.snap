// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders content 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="emotion-0"
    data-testid="flex"
  >
    The SMS program is down, synthesize the redundant feed so we can synthesize the SQL hard drive!
  </div>
</div>
`;
